go/vt/vtgate/executor.go

```diff
@@ -812,277 +655,101 @@ func (e *Executor) handleShowVitessMetadata(ctx context.Context, opt *sqlparser.
 	}, nil
 }
 
-func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, dest key.Destination, destKeyspace string, destTabletType topodatapb.TabletType, logStats *LogStats) (*sqltypes.Result, error) {
-	stmt, err := sqlparser.Parse(sql)
-	if err != nil {
-		return nil, err
-	}
-	showOuter, ok := stmt.(*sqlparser.Show)
-	if !ok {
-		// This code is unreachable.
-		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "[BUG] unrecognized SHOW statement: %v", sql)
-	}
-	show, ok := showOuter.Internal.(*sqlparser.ShowLegacy)
-	if !ok {
-		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "[BUG] This should only be SHOW Legacy statement type: %v", sql)
-	}
-	ignoreMaxMemoryRows := sqlparser.IgnoreMaxMaxMemoryRowsDirective(stmt)
-	execStart := time.Now()
-	defer func() { logStats.ExecuteTime = time.Since(execStart) }()
-	switch strings.ToLower(show.Type) {
-	case sqlparser.KeywordString(sqlparser.VARIABLES):
-		if show.Scope == sqlparser.VitessMetadataScope {
-			return e.handleShowVitessMetadata(ctx, show.ShowTablesOpt)
-		}
-	// for ENGINES, we want to return just InnoDB
-	case sqlparser.KeywordString(sqlparser.ENGINES):
-		rows := make([][]sqltypes.Value, 0, 6)
-		row := buildVarCharRow(
-			"InnoDB",
-			"DEFAULT",
-			"Supports transactions, row-level locking, and foreign keys",
-			"YES",
-			"YES",
-			"YES")
-		rows = append(rows, row)
-		return &sqltypes.Result{
-			Fields: buildVarCharFields("Engine", "Support", "Comment", "Transactions", "XA", "Savepoints"),
-			Rows:   rows,
-		}, nil
-	// for PLUGINS, return InnoDb + mysql_native_password
-	case sqlparser.KeywordString(sqlparser.PLUGINS):
-		rows := make([][]sqltypes.Value, 0, 5)
-		row := buildVarCharRow(
-			"InnoDB",
-			"ACTIVE",
-			"STORAGE ENGINE",
-			"NULL",
-			"GPL")
-		rows = append(rows, row)
-		return &sqltypes.Result{
-			Fields: buildVarCharFields("Name", "Status", "Type", "Library", "License"),
-			Rows:   rows,
-		}, nil
-	case sqlparser.KeywordString(sqlparser.VITESS_SHARDS):
-		showVitessShardsFilters := func(show *sqlparser.ShowLegacy) ([]func(string) bool, []func(string, *topodatapb.ShardReference) bool) {
-			keyspaceFilters := []func(string) bool{}
-			shardFilters := []func(string, *topodatapb.ShardReference) bool{}
-
-			if show.ShowTablesOpt == nil || show.ShowTablesOpt.Filter == nil {
-				return keyspaceFilters, shardFilters
-			}
-
-			filter := show.ShowTablesOpt.Filter
+// (tablet, servingState, mtst) -> bool
```

Comment:
nit: you could add names to the tabletFilter arguments below instead of commenting